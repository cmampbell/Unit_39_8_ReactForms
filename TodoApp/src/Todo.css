.Todo {
    display: inline-flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
    width: 300px;
}

.Todo-text {
    font-weight: bold;
    margin-right: 20px;
}

.Todo-delete {
    background-blend-mode: lighten;
    background-color: rgba(253, 253, 253, 0);
    border: none;
    border-radius: 5px;
}

.Todo-delete:hover {
    background-color: rgba(253, 253, 253, 0.315);;
}