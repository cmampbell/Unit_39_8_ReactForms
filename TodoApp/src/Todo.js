import React from "react";
import "./Todo.css"

const Todo = ({text, deleteTodo}) => (
    <div className="Todo">
        <p className="Todo-text">{text}</p>
        <button className="Todo-delete"onClick={() => deleteTodo(text)}>X</button>
    </div>
)

export default Todo;