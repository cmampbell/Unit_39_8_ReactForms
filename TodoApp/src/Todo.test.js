import React from 'react';
import { render } from "@testing-library/react";
import Todo from './Todo';


it('renders without crashing', function() {
    render(<Todo text="Test"/>);
});

it('matches snapshot', function() {
    const {asFragment} = render(<Todo text="Test"/>);
    expect(asFragment()).toMatchSnapshot();
});