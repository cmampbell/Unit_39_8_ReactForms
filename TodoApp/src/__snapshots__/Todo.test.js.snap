// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Todo"
  >
    <p
      class="Todo-text"
    >
      Test
    </p>
    <button
      class="Todo-delete"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
